pkg/core/tekton/builder/pipeline.go

```diff
@@ -0,0 +1,131 @@
+package builder
+
+import (
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PipelineBuilder holds the tekton pipeline definition.
+type PipelineBuilder struct {
+	Pipeline v1beta1.Pipeline
+}
+
+// NewPipelineBuilder creates a Pipeline with default values.
+func NewPipelineBuilder(name, namespace string) *PipelineBuilder {
+	b := &PipelineBuilder{}
+	b.Pipeline = v1beta1.Pipeline{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	return b
+}
+
+// PipelineMeta sets the Meta structs of the Pipeline.
+// Any number of MetaOp modifiers can be passed.
+func (b *PipelineBuilder) PipelineMeta(ops ...MetaOp) {
+	for _, op := range ops {
+		switch o := op.(type) {
+		case ObjectMetaOp:
+			o(&b.Pipeline.ObjectMeta)
+		case TypeMetaOp:
+			o(&b.Pipeline.TypeMeta)
+		}
+	}
+}
+
+// PipelineDescription sets the Description on the Pipeline spec.
+func (b *PipelineBuilder) PipelineDescription(description string) {
```

Comment:
A best practice with golang naming is to strip from the method names the tokens that are already reflected in the type to which the methods belong. In this case, is it really necessary to use `Pipeline` as a prefix to all these methods, or is it already implied from the parent `PipelineBuilder` type that all these methods apply to a pipeline ?

Reply:
Indeed adding `Pipeline` makes it too verbose without any value.